### ISSUE TITLE
False positive: shellcheck wrongly accuse "type" not in POSIX (SC2039)

### ISSUE THREAD

**Issue by congma** [koalaman/shellcheck#899]
#### For bugs

- Rule Id (if any, e.g. SC1000):  `SC2039`
- My shellcheck version (`shellcheck --version` or "online"):  both `0.4.6` and `online`
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
-  <s>It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue</s> It is reproducible online.

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
type cd
```

#### Here's what shellcheck currently says:

```
Line 2:
type cd
^-- SC2039: In POSIX sh, 'type' is undefined.
```

#### Here's what I wanted or expected to see:

Nothing, for `type` is indeed defined in POSIX.  See: `man 1p type` (assuming POSIX man pages installed), or [The Open Group documentation](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/type.html#top).

#### Additional comments:

Relevant line: https://github.com/koalaman/shellcheck/blob/324aa3cc88eefb79a24580e7746801e096e09332/ShellCheck/Checks/ShellSupport.hs#L282

**Comment by koalaman** [comment]
It's defined as an XSI extension, it's not available in base POSIX.

**Comment by congma** [comment]
Yes. Somehow I missed this.
